packages/application-shell/src/components/quick-access/index.js

```diff
@@ -36,11 +42,18 @@ class QuickAccessContainer extends React.Component {
   }
 }
 
-export default class QuickAccessTrigger extends React.Component {
+export class QuickAccessTrigger extends React.Component {
   static displayName = 'QuickAccessTrigger';
 
+  static propTypes = {
+    getPimSearchStatusForProject: PropTypes.func.isRequired,
+  };
+
   componentDidMount() {
     document.addEventListener('keydown', this.handler);
+    if (this.props.project) {
+      this.updatePimSearchInfo();
+    }
```

Comment:
We check whether PIM-Search is enabled first to avoid making requests to pim-search for projects which don't use pim-search yet.

Reply:
> Can we maybe check this only when the user starts searching for products specifically?

This is not possible. We do not know what the user is searching for. They just enter a text and we find anything that could match. So we always need to search products.

> I'm just wondering if we need to do this 

We use this information to avoid making requests when a user searches. If the project is not indexed then we don't search products using pim-search at all and don't make a request again. We check it once upfront so that we already know whether we need to query pim-search when the user actually starts typing.

In the beginning I loaded the information when the search component itself mounts, but that felt wrong as it would lead to a new request every time quick-access is opened again (as the component loses state when it unmounts).

> I'm not too familiar with the internals, so I can't say for sure if that would be possible...

I think we can get the best of both worlds. We can store the information of whether to search or not in the state of `QuickAccessTrigger` (where it is now), and we can load that state (in case it was not loaded yet) when `QuickAccess` mounts.

This means we:
- load the information only once (per application)
- load the information only when the user activates quick-access instead of always

Thanks for pointing it out 👍

Reply:
There is no good caching in `sdk`. Moving the request to `cDM` of `QuickAccess` seems ideal as that's exactly when we need the information. By letting the parent handle the state we avoid making the request twice and we control the "caching" ourselves. I'm sold on that approach as it's simple and minimal effort.

Reply:
I'd like to have at least a code comment explaining those trade-offs.

Reply:
I can add that once it's implemented 👍

Reply:
Done in the latest commit 40d576e598d31c9740e9791d2ab2abd7fbe4c8e8